internal/dinosaur/pkg/services/quota/ams_quota_service.go

```diff
@@ -158,6 +165,27 @@ func (q amsQuotaService) ReserveQuota(dinosaur *dbapi.CentralRequest, instanceTy
 	return "", errors.InsufficientQuotaError("Insufficient Quota")
 }
 
+func (q amsQuotaService) verifyCloudAccountInAMS(dinosaur *dbapi.CentralRequest, orgID string) *errors.ServiceError {
+	cloudAccounts, err := q.amsClient.GetCustomerCloudAccounts(orgID, []string{RHACSMarketplaceQuotaID})
+	if err != nil {
+		svcErr := errors.ToServiceError(err)
+		return errors.NewWithCause(svcErr.Code, svcErr, "Error getting cloud accounts")
+	}
+
+	if dinosaur.CloudAccountID == "" && len(cloudAccounts) != 0 {
+		return errors.CloudAccountIDNotSetupProperly("Request cloud account is not setup even though organization has cloud accounts")
+	}
+	if dinosaur.CloudAccountID == "" && len(cloudAccounts) == 0 {
+		return nil
+	}
```

Comment:
I would probably write this as such to avoid the `&&`s. But feel free to ignore, I think this is personal taste.

```suggestion
	if dinosaur.CloudAccountID == "" {
	  if len(cloudAccounts) != 0 {
		return errors.CloudAccountIDNotSetupProperly("Request cloud account is not setup even though organization has cloud accounts")
	  }
	  return nil
}
```